crates/commonware-node/src/dkg/manager/export.rs

```diff
@@ -0,0 +1,44 @@
+//! Export functions for DKG signing share migration.
+//!
+//! This module provides functionality to export the signing share at epoch boundaries,
+//! enabling coordinated network upgrades with breaking storage changes.
+
+use std::path::Path;
+
+use commonware_cryptography::bls12381::primitives::group::Share;
+use eyre::{Result, WrapErr as _};
+use tempo_commonware_node_config::SigningShare;
+use tracing::info;
+
+/// Write the signing share to a file in hex format.
+///
+/// The exported file can be used directly as `--consensus.signing-share` for a new node.
+/// If the file already exists, this is a no-op to prevent accidental overwrites.
+pub fn write_share_to_file(share: &Share, path: &Path) -> Result<()> {
+    // Check if file exists to avoid overwriting
+    if path.exists() {
```

Comment:
IMO this check should not exist - we should just blast over the target file. Otherwise you end up in a weird spot where your binary runs for a long time and then you end up not writing. I guess you could re-run the binary, but somehow this does not feel right.

Alternative suggestion: just write out the signing share on every new DKG result?